src/main/kotlin/systems/ajax/malov/stockanalyzer/repository/StockRecordRepository.kt

```diff
@@ -1,10 +1,12 @@
 package systems.ajax.malov.stockanalyzer.repository
 
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 import systems.ajax.malov.stockanalyzer.entity.MongoStockRecord
 import java.util.Date
 
 interface StockRecordRepository {
-    fun insertAll(mongoStockRecords: List<MongoStockRecord>): List<MongoStockRecord>
-    fun findTopNStockSymbolsWithStockRecords(n: Int, from: Date, to: Date): Map<String, List<MongoStockRecord>>
-    fun findAllStockSymbols(): List<String>
+    fun insertAll(mongoStockRecords: List<MongoStockRecord>): Flux<MongoStockRecord>
+    fun findTopNStockSymbolsWithStockRecords(n: Int, from: Date, to: Date): Mono<Map<String, List<MongoStockRecord>>>
```

Comment:
why do we need to return `Map`?

`Mono<Map<String, List<MongoStockRecord>>>` looks ugly

Reply:
Business logic